src/App.tsx

```diff
@@ -4,13 +4,17 @@ import { useChat } from './useChat';
 import { ChatInterface } from './ChatInterface';
 import { useIsMobileLayout, SearchHistoryModal, SideBar } from 'lib-kava-ai';
 import hardAILogo from './assets/hardAILogo.svg';
+import WalletModal from './WalletModal';
+import { PROMOTED_WALLETS } from './utils/wallet';
+import { WalletProviderDetail } from './types';
 
 const sideBarLogo = <img src={hardAILogo} alt="Hard AI logo" height={18} />;
 
 export const App = () => {
   const [isMobileSideBarOpen, setIsMobileSideBarOpen] = useState(false);
   const [isDesktopSideBarOpen, setIsDesktopSideBarOpen] = useState(true);
   const [isSearchHistoryOpen, setIsSearchHistoryOpen] = useState(false);
+  const [isWalletConnectOpen, setIsWalletConnectOpen] = useState(false);
```

Comment:
this being the second modal is making me wonder if we should consider building a modal context